### PR TITLE
internal/docs: Replace "user-facing type" and "concrete type" with "portable type"

### PR DESCRIPTION
**Pull request by clausti**
To date, the documentation has used "user-facing type" and "concrete type" fairly interchangeably, but having more precise terminology will be helpful for user communication and the CLI documentation. 

updates https://github.com/google/go-cloud/issues/1470

### CODE REVIEW

File: internal/docs/design.md [hunk 1]
```diff
@@ -130,18 +131,18 @@ symbols stable over time.
 The exception to this rule is if the name is not unique across providers. The
 canonical example is `gcpkms` and `awskms`.
 
-## Concrete Type Constructors
+## Portable Type Constructors
 
-Concrete type constructors are the functions defined in provider-specific
-packages that end users use to get an instance of the concrete type. For
+Portbale type constructors are the functions defined in provider-specific
```
Comment (zombiezen) [inline internal/docs/design.md]: Spelling:

```suggestion
Portable type constructors are the functions defined in provider-specific
```

File: internal/docs/design.md [hunk 2]
```diff
@@ -130,18 +131,18 @@ symbols stable over time.
...
+packages that end users call to get an instance of the portable type. For
 example, `gcsblob.OpenBucket`, which returns an instance of the `*blob.Bucket`
-concrete type backed by GCS.
+portbale type backed by GCS.
```
Comment (zombiezen) [inline internal/docs/design.md]: Same here:

```suggestion
portable type backed by GCS.
```

File: internal/docs/design.md [hunk 3]
```diff
@@ -364,9 +365,9 @@ Driver implementations should:
 -   Return the raw errors from the underlying provider, and not wrap them in
     `fmt.Errorf` calls, so that they can be exposed to end users via `ErrorAs`.
 
-### Concrete Types
+### Portble Types
```
Comment (zombiezen) [inline internal/docs/design.md]:
```suggestion
### Portable Types
```

**Comment by vangent** [comment]
Is the godoc in the code also consistent with this? And our README.md, gocloud.dev, etc...? Our documentation should be consistent across the board.

**Comment by clausti** [comment]
@vangent that is a great point, and if the nomenclature looks good to you/folks, I can make this PR a more comprehensive update.

**Comment by clausti** [comment]
Also, to all y'all: I have now stared at the word "portable" for so long I can't tell if it's spelled correctly or even a word at all so typo flags are appreciated.

**Comment by eliben** [comment]
+1 for this terminology alignment

We should also plan to move this part of the documentation from design.md into something more user-visible like doc.go or even the landing page, but all of that should be done in a separate PR

**Comment by clausti** [comment]
ok, re: @vangent's comment about updating everything: https://github.com/google/go-cloud/issues/1470

and re: @eliben's comment about moving information to a better place: https://github.com/google/go-cloud/issues/1469

**Pull request merged**